ReSharper.FSharp/src/FSharp.Psi.Services/src/Generate/GenerateProvider.fs

```diff
@@ -236,15 +238,28 @@ type FSharpOverridingMembersBuilder() =
                 let diff = m.Indent - indentationTarget
                 if diff > 0  && isFirstMeaningfulNodeOnLine m then shiftWithWhitespaceBefore -diff m)
 
-        // members etc. might need indenting or deindenting after the representation start was moved to it's own line
         let reprNodes =
-            typeRepr.Children()
-            |> Seq.filter
-                   (fun c -> c :? ICommentNode || c :? ITypeBodyMemberDeclaration || c :? IRecordFieldDeclarationList)
+            let children = typeRepr.Children() |> Seq.toList
+            let startIdx = children |> List.tryFindIndex (fun c -> c.NodeType = FSharpTokenType.CLASS
+                                                                    || c.NodeType = FSharpTokenType.STRUCT
+                                                                    || c.NodeType = FSharpTokenType.LBRACE)
+            let endIdx =
+                children
+                |> List.tryFindIndexBack
+                       (fun c -> c.NodeType = FSharpTokenType.END || c.NodeType = FSharpTokenType.RBRACE)
+            match (startIdx, endIdx) with
+            | Some s, Some e when s >= 0 && e > s ->
+                let first, last = s + 1, e - 1
+                children[first .. last] |> List.filter (isWhitespace >> not)
+            | _ -> List.empty
         deindent (desiredIndent + indentSize) reprNodes
 
         let declNodes =
-            typeDecl.Children() |> Seq.filter (fun c -> c :? ICommentNode || c :? ITypeBodyMemberDeclaration)
+            let children = typeDecl.Children() |> Seq.toList
+            let idx = children |> List.tryFindIndex(fun c -> c :? ITypeRepresentation)
+            match idx with
+            | Some idx when idx >= 0 -> List.skip (idx + 1) children |> List.filter (isWhitespace >> not)
```

Comment:
`Seq.skipWhile (function | :? ITypeRepresentation -> false | _ -> true)`
or
`Seq.skipWhile ((!=) typeRepr)`

Reply:
> `List.filter (isWhitespace >> not)`

I think you should also update the whitespace nodes, at least on lines where whitespaces are the only nodes on a line.

There's a whitespace of length 12 on the line between the fields here:
```fsharp
type R = { F1: int
           
           F2: int }
```

It's not being updated here:
```fsharp
type R =
    { F1: int
           
      F2: int }
```

Reply:
Okay, as the empty line was handled as part of the parent node I introduced some traversing to get to the empy-line children.